Data/models/DataSaved.swift

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import CoreData
+import Shared
+
+private let log = Logger.browserLogger
+
+public final class DataSaved: NSManagedObject, CRUD {
+    @NSManaged public var savedUrl: String
+    @NSManaged public var amount: String
+    
+    public class func get(with savedUrl: String) -> DataSaved? {
+        let predicate = NSPredicate(format: "\(#keyPath(DataSaved.savedUrl)) == %@", savedUrl)
+        return first(where: predicate)
+    }
+    
+    public class func all() -> [DataSaved] {
+        all() ?? []
+    }
+    
+    public class func delete(with savedUrl: String) {
+        if let item = get(with: savedUrl) {
```

Comment:
There is a threading conflict, you fetch the item on main context, then delete it on background context.

What you should do to fetch the item on background context too, and delete it on the same context
so under `DataController.perform` you do the fetch

You also have to split `get()` function to public and internal implementations, public function stays the same,
while to `getInternal` you can pass the context you just created in `DataController.perform` method
you can look for `TabMO.getInternal` implementation for example

Reply:
Internal functionality is added and context problems are resolved.